features/dnsresolver/netty/src/main/java/org/opennms/netmgt/dnsresolver/netty/RandomIterator.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.dnsresolver.netty;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * An infinite iterator that will return random items
+ * from the given list.
+ *
+ * @param <T> the type of elements in the list
+ */
+public class RandomIterator<T> implements Iterable<T> {
+    private final List<T> items;
+    private final int numItems;
+    private final Random random;
+
+    public RandomIterator(final List<T> coll) {
+        items = new ArrayList<>(coll);
+        numItems = items.size();
+        random = new Random();
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new Iterator<T>() {
+            @Override
+            public boolean hasNext() {
+                return true;
+            }
+
+            @Override
+            public synchronized T next() {
+                return items.get(random.nextInt(numItems));
+            }
+
+            @Override
+            public void remove() {
+                throw new IllegalArgumentException("Cannot remove from " + RandomIterator.class);
```

Comment:
An IllegalArgumentException refers to a wrong parameter. I think an UnsupportedOperationException is a better choice here.

Reply:
Good point.